internal/services/kusto/kusto_cosmosdb_data_connection_resource.go

```diff
@@ -0,0 +1,248 @@
+package kusto
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/kusto/2022-12-29/dataconnections"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/kusto/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type CosmosDBDataConnectionModel struct {
+	Name               string `tfschema:"name"`
+	ResourceGroupName  string `tfschema:"resource_group_name"`
+	Location           string `tfschema:"location"`
+	CosmosDbAccountId  string `tfschema:"cosmosdb_account_id"`
+	CosmosDbContainer  string `tfschema:"cosmosdb_container"`
+	CosmosDbDatabase   string `tfschema:"cosmosdb_database"`
+	ClusterName        string `tfschema:"cluster_name"`
+	DatabaseName       string `tfschema:"database_name"`
+	ManagedIdentityId  string `tfschema:"managed_identity_id"`
+	MappingRuleName    string `tfschema:"mapping_rule_name"`
+	RetrievalStartDate string `tfschema:"retrieval_start_date"`
+	TableName          string `tfschema:"table_name"`
+}
+
+var _ sdk.Resource = CosmosDBDataConnectionResource{}
+
+type CosmosDBDataConnectionResource struct{}
+
+func (r CosmosDBDataConnectionResource) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.DataConnectionName,
+		},
+		"resource_group_name": commonschema.ResourceGroupName(),
+		"location":            commonschema.Location(),
+		"cosmosdb_account_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: azure.ValidateResourceID,
+		},
+		"cosmosdb_container": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+		"cosmosdb_database": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+		"cluster_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ClusterName,
+		},
+		"database_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.DatabaseName,
+		},
+		"managed_identity_id": {
```

Comment:
If this is the managed identity from the cosmosdb resource can we specify that in the name?
```suggestion
		"cosmosdb_managed_identity_id": {
```

Reply:
not exactly, this identity id could be user assigned identity id or a system assigned identity id  (resource id) in some cases.

Reply:
You could see the similar managed_identity_id in other kusto data connection resources